tree/ntuple/v7/doc/BinaryFormatSpecification.md

```diff
@@ -802,6 +802,27 @@ This section is a comprehensive list of the C++ types with RNTuple I/O support.
 Within the supported type system complex types can be freely composed,
 e.g. `std::vector<MyEvent>` or `std::vector<std::vector<float>>`.
 
+### Type Name Normalization
+
+Type names are stored according to the following normalization rules
+  - All whitespaces are removed.
+  - CV qualifiers (`const`, `volatile`) are removed.
+  - The `class`, `struct`, and `enum` keywords are removed.
+  - Type names are fully qualified by the namespace in which they are declared.
+  - The integer types `signed char`, `unsigned char`, and `[signed|unsigned](short|int|long[ long])` are replaced by
+    the corresponding (at the time of writing) `std::[u]int(8|16|32|64)_t` standard integer typedef
+  - Type names used as template arguments are themselves normalized.
+  - For user-defined types, default-initialized template arguments are explicitly spelled out;
+    optional template arguments of stdlib types such as allocators are omitted.
+  - Integer template arguments are written in standard decimal notation; only if necessary (long integer types),
+    a suffix `l` or `ul` is added according to the template parameter type.
+
+The type name of a field has typedefs and usings fully resolved (except for standard integer typedefs).
```

Comment:
This may belong sooner in the document as the above mentioned rules are applying to the name with fully resolved typedefs.

Reply:
Hm, I'm not sure I understand. The normalization rules, as far as I understand, should apply to both the original type name and the resolved type name.

Reply:
The replacement of the typedef is an integral type of the 'normalization' of the name.  Without that part, the name are not unique and thus not normalized.  I.e. if you have both `const vector<vector<Int_t> >` and `vector<vector<int32_t, typedef_to_regular_allocator>>` "only" pretty printing the name does not help much.  Similar if the typedef are not replaced, neither would/should the Integer template arguments be calculated/compacted.  In the second example, without resolving the typedef we can not even remove the default template argument.

Reply:
If the user passes us `vector<vector<int32_t, typedef_to_regular_allocator>>`, its "pretty printed" version will be registered as the field's alias type, i.e. the "original spelling" from the user. Otherwise (`vector<vector<int>>`), the alias type is empty.

Reply:
For the original spelling, I guess we ought to decide to either really put it as-is or we need to spell out, separately from the normalization, what applies there.
For example, it is clear that the 'whitespace' rule would apply,  maybe the `const` rule and what else?  (Not the `typedef` for sure, what about the `std::[u]int(8|16|32|64)_t` or even the fully qualified)